### PR TITLE
Add GitHub Workflow Replacement for Jenkins job_PostCommit_Java_Examples_Dataflow_Java11

### PR DESCRIPTION
**Pull request by aleksandr-dudko**
The following jenkins job has been migrated:

https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_Java_Examples_Dataflow_Java11/

How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

by adding a comment to the PR

image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PostCommit_Java_Dataflow_Examples_Java11.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          java-version: '11'
+      - name: run PostCommit Java script
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+            gradle-command: :runners:google-cloud-dataflow-java:examples:java11PostCommit
+            arguments: |
+              -PcompileAndRunTestsWithJava11 \
+              -Pjava11Home=$JAVA_HOME_11_X64 \
+            default-arguments: |
+              --continue \
```
[the remark this answers is not in the document]
Comment (aleksandr-dudko) [reply .github/workflows/beam_PostCommit_Java_Dataflow_Examples_Java11.yml:89]: Here is the example of Jenkins gradle task:
`22:01:07 [src] $ /home/jenkins/jenkins-slave/workspace/beam_PostCommit_Java11_Examples_Dataflow/src/gradlew --continue --max-workers=6 -Dorg.gradle.jvmargs=-Xms2g -Dorg.gradle.jvmargs=-Xmx4g :runners:google-cloud-dataflow-java:examples:java11PostCommit`

File: .github/workflows/beam_PostCommit_Java_Examples_Dataflow_Java11.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    name: ${{matrix.job_name}} (${{matrix.job_phrase}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 180
+    strategy:
+      fail-fast: false
+      matrix:
+        job_name: [beam_PostCommit_Java_Examples_Dataflow_Java11]
+        job_phrase: [Run Java examples on Dataflow Java 11]
+    if: |
+      github.event_name == 'push' ||
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PostCommit_Java_Examples_Dataflow_Java11.yml:64]: There is no `push` event for PostCommit jobs

Comment (aleksandr-dudko) [reply .github/workflows/beam_PostCommit_Java_Examples_Dataflow_Java11.yml]: It's done. Thank you.

**Review by magicgoody — APPROVED** [review]

**Review by andreydevyatkin — APPROVED** [review]
LGTM

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**